crypto/src/main/java/org/tron/common/crypto/SignUtils.java

```diff
@@ -26,10 +26,17 @@ public static SignInterface fromPrivate(byte[] privKeyBytes, boolean isECKeyCryp
   public static byte[] signatureToAddress(
       byte[] messageHash, String signatureBase64, boolean isECKeyCryptoEngine)
       throws SignatureException {
-    if (isECKeyCryptoEngine) {
-      return ECKey.signatureToAddress(messageHash, signatureBase64);
+    try {
+      if (isECKeyCryptoEngine) {
+        return ECKey.signatureToAddress(messageHash, signatureBase64);
+      }
+      return SM2.signatureToAddress(messageHash, signatureBase64);
+    } catch(IllegalArgumentException e) {
+      if (e.getMessage().contains("Invalid point compression")) {
```

Comment:
Why special treatment, I understand that you can throw a SignatureException whenever `signatureToAddress` is incorrect.

Reply:
Currently only the exception with message like "Invalid point compression" can be sure to deal like this way, and other kind of messages can not be found now, that means just treat it like this may be not correct and that will not be noticed in this way.

Reply:
I'm not sure I agree with this.

Reply:
From the method signatureToKeyBytes in class ECKey, other kinds of error messages has been judged, that means not treat them the same, and it is core method ,any error messages deserved to be noticed, if we are not sure how to deal, we should throw it.

Reply:
> IllegalArgumentException

When the signature is not correct but the length is right, for example, if someone send a transaction , and the correct signature is "abc", but it was replaced to "001", that will bring it a IllegalArgumentException.

Reply:
> Actually, I think all exceptions in this scene are SignatureException even `001`

yes , it is clear to throw an SignatureException in this scene, while the point is that  shall we just throw a SignatureException in any unknown scene. @tomatoishealthy

Reply:
updated.